### PR TITLE
[namespace] Move disptacher to namespace

### PR DESCRIPTION
**Pull request by laoneo**
The disptacher functionality introduced with #13687 as namespaced version.

### CODE REVIEW

File: libraries/src/Cms/Dispatcher/Dispatcher.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	/**
+	 * Constructor for Dispatcher
+	 *
+	 * @param   \JApplicationCms  $app       The JApplication for the dispatcher
+	 * @param   \JUser            $user      The user for the dispatcher
+	 * @param   \JLanguage        $language  The language for the dispatcher
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function __construct(\JApplicationCms $app, \JUser $user, \JLanguage $language)
```
Comment (wilsonge) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: There's no need for these extra properties. `$app->getUser()` will return the user object, and `$app->getLanguage` will get the language object

**Review by wilsonge — CHANGES_REQUESTED** [review]

File: libraries/src/Cms/Dispatcher/Dispatcher.php (hunk 1 above)
Comment (wilsonge) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: You are also missing these properties when you create the class in `JComponentHelper` so banners is now fatal erroring.

**Comment by wilsonge** [comment]
👍 for namespacing - but your modifications are wrong :)

File: libraries/src/Cms/Dispatcher/Dispatcher.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+abstract class Dispatcher implements DispatcherInterface
+{
+	/**
+	 * The JApplication instance
+	 *
+	 * @var     \JApplicationCms
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private $app;
```
Comment (joomdonation) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: Could I ask why $app is private? What if in component dispatcher (like in BannersDispatcher class) we need to access to app object ? Maybe it should be protected instead of private?

File: libraries/src/Cms/Dispatcher/Dispatcher.php (hunk 1 above)
Comment (laoneo) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: getUser() is not available

File: libraries/src/Cms/Dispatcher/Dispatcher.php (hunk 2 above)
Comment (laoneo) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: I would even pass the app as argument to the dispatch function. But don't know why it was decided as a constructor argument, so I left it as it was made in the original PR.

File: libraries/src/Cms/Dispatcher/Dispatcher.php (hunk 1 above)
Comment (wilsonge) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: Apologies I meant `getIdentity()` (which comes from https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Cms/Application/IdentityAware.php)

**Comment by laoneo** [comment]
My last commit is using the user and language from the app and making the app available trough `getApplication()`.

**Comment by wilsonge** [comment]
:+1:

File: administrator/components/com_installer/views/manage/view.html.php [hunk 3]
```diff
@@ -24,8 +24,6 @@ class InstallerViewManage extends InstallerViewDefault
 
 	protected $form;
 
-	protected $state;
```
Comment (wilsonge) [inline administrator/components/com_installer/views/manage/view.html.php]: Why are we remove the state here?

File: libraries/src/Cms/Dispatcher/Dispatcher.php (hunk 2 above)
Comment (wilsonge) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: Originally because I was going to load language files in the constructor. But @mbabker objected. So there's no good reason. But it probably does belong here anyhow, so extensions doing any other logic can do thing in the constructor

**Comment by wilsonge** [comment]
If you can remove this com_installer code i guess you accidently committed this is good to go

File: administrator/components/com_installer/views/manage/view.html.php (hunk 3 above)
Comment (laoneo) [inline administrator/components/com_installer/views/manage/view.html.php]: As I was testing to install DPCalendar I got an error that $state needs to be public. It is already set up in the base class, so no need to do it here again, that's why I removed it here. It's a bug fix.

Comment (wilsonge) [inline administrator/components/com_installer/views/manage/view.html.php]: This isn't a new property in 4.x. This needs to go back into staging if it's a bug. Or at least put it into a separate PR. Let's keep PR's single responsibility

Comment (laoneo) [inline administrator/components/com_installer/views/manage/view.html.php]: This is introduced in 4 https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_installer/views/default/view.php#L25 and not available in https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_installer/views/default/view.php.

Comment (wilsonge) [inline administrator/components/com_installer/views/manage/view.html.php]: Right then that comes from this https://github.com/joomla/joomla-cms/pull/12173 let's open up a separate issue for that. Because that's totally unrelated to dispatcher and so let's handle it in a separate PR.

**Comment by laoneo** [comment]
Reverted the change. Will do a pr with the cleanup.

**Comment by wilsonge** [comment]
:+1:

File: administrator/components/com_installer/views/manage/view.html.php (hunk 3 above)
Comment (laoneo) [inline administrator/components/com_installer/views/manage/view.html.php]: Done with #13980.
